src/main/java/org/myrobotlab/service/Python.java

```diff
@@ -871,54 +871,91 @@ public void stopService() {
     // release the interpeter
     stop();
   }
+  
+  public void init() {
+    
+    log.info("created python {}", getName());
+    createPythonInterpreter();
+    sleep(250);
+    
+    inputQueueThread = new InputQueue(this);
 
-  public boolean isOpenOnExecute() {
-    return openOnExecute;
-  }
-
-  public void setOpenOnExecute(boolean openOnExecute) {
-    this.openOnExecute = openOnExecute;
-  }
+    // I love ServiceData !
+    ServiceData sd = ServiceData.getLocalInstance();
+    List<MetaData> sdt = sd.getAvailableServiceTypes();
+    for (int i = 0; i < sdt.size(); ++i) {
+      MetaData st = sdt.get(i);
+      // FIXME - cache in "data" dir Or perhaps it should be pulled into
+      // resource directory during build time and packaged with jar
+      String file = String.format("%s/%s.py", st.getSimpleName(), st.getSimpleName());
+      exampleFiles.put(st.getSimpleName(), file);
+    }
 
-  public static void main(String[] args) {
-    try {
-      LoggingFactory.init("INFO");
+    localPythonFiles = getFileListing();
 
-      // Runtime.start("i01.head.rothead", "Servo");
-      // Runtime.start("i01.head.neck", "Servo");
-      WebGui webgui = (WebGui) Runtime.create("webgui", "WebGui");
-      webgui.autoStartBrowser(false);
-      webgui.startService();
-      Python python = (Python) Runtime.start("python", "Python");
-      // python.execFile("data/adafruit.py");
+    attachPythonConsole();
 
-      // Runtime.start("i01", "InMoov2");
+    String selfReferenceScript = "from time import sleep\nfrom org.myrobotlab.framework import Platform\n" + "from org.myrobotlab.service import Runtime\n"
+        + "from org.myrobotlab.framework import Service\n" + "from org.myrobotlab.service import Python\n"
+        + String.format("%s = Runtime.getService(\"%s\")\n\n", CodecUtils.getSafeReferenceName(getName()), getName()) + "Runtime = Runtime.getInstance()\n\n"
+        + String.format("runtime = Runtime.getInstance()\n") + String.format("myService = Runtime.getService(\"%s\")\n", getName());
+    // FIXME !!! myService is SO WRONG it will collide on more than 1 python

```

Comment:
Is the `myService` variable still needed? As far as I've seen it's never used and instead the `python` variable or service is used

Reply:
really it should start only with runtime ... this is the same behavior as py4j .. ie runtime reference is all you need to get things started ...
Will remove..

Reply:
ah .. it is needed, its need to handle std:out from the jython ... kind of a critical piece 
This is the same for Py4j, in general the python needs to know the service name to hijack the std:out std:err of the python
In Py4j I had to create a setName on the Executor interface so the Python process can know which mrl service is responsible for creating it.